### ISSUE TITLE
Crash when moving a pinned floating window to special workspace

### ISSUE THREAD

**Issue by foonathan** [hyprwm/Hyprland#2395]
## Steps to reproduce

1. Open a terminal emulator (I'm using `kitty`, but don't think it matters).
2. Make the window floating and pin.
3. Inside that terminal, run `hyprctl dispatch movetoworkspace silent special`. Nothing appears to happen.
4. Quickly spam `CTRL+D` inside the terminal until hyprland crashes.

## Crash log

```
--------------------------------------------
   Hyprland Crash Report
--------------------------------------------
This was an accident, I swear!

Hyprland received signal 11 (Segmentation fault)

Version: f27873a6f06dc2f87600edb890f3c38298bfb55f

System info:
	System name: Linux
	Node name: foonathan-x1
	Release: 6.3.4-arch1-1
	Version: #1 SMP PREEMPT_DYNAMIC Wed, 24 May 2023 17:44:00 +0000

GPU:
	00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] [8086:9a49] (rev 01) (prog-if 00 [VGA controller])


os-release:
	NAME="Arch Linux"
	PRETTY_NAME="Arch Linux"
	ID=arch
	BUILD_ID=rolling
	ANSI_COLOR="38;2;23;147;209"
	HOME_URL="https://archlinux.org/"
	DOCUMENTATION_URL="https://wiki.archlinux.org/"
	SUPPORT_URL="https://bbs.archlinux.org/"
	BUG_REPORT_URL="https://bugs.archlinux.org/"
	PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
	LOGO=archlinux-logo
	


Backtrace:
	#0 | Hyprland(_ZN13CrashReporter18createAndSaveCrashEi+0x450) [0x555939ea0b20]
		??
		??:0
	#1 | Hyprland(_Z25handleUnrecoverableSignali+0x47) [0x555939e41667]
		??
		??:0
	#2 | /usr/lib/libc.so.6(+0x39ab0) [0x7fb388962ab0]
		??
		??:0
	#3 | Hyprland(+0x85084) [0x555939e3f084]
		??
		??:0
	#4 | Hyprland(_ZN13CHyprRenderer28renderAllClientsForWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK8Vector2DRKf+0x95b) [0x555939f2611b]
		??
		??:0
	#5 | Hyprland(_ZN13CHyprRenderer15renderWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK7wlr_box+0xde) [0x555939f27b6e]
		??
		??:0
	#6 | Hyprland(_ZN13CHyprRenderer13renderMonitorEP8CMonitor+0x5e2) [0x555939f2a342]
		??
		??:0
	#7 | Hyprland(_Z13handleWrappedP11wl_listenerPv+0x40) [0x555939ecbe30]
		??
		??:0
	#8 | /usr/lib/libwayland-server.so.0(wl_signal_emit_mutable+0x7e) [0x7fb3892a801e]
		??
		??:0
	#9 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch_idle+0x23) [0x7fb3892a9a23]
		??
		??:0
	#10 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0xfd) [0x7fb3892a9b3d]
		??
		??:0
	#11 | /usr/lib/libwayland-server.so.0(wl_display_run+0x27) [0x7fb3892aa2d7]
		??
		??:0
	#12 | Hyprland(main+0x4a7) [0x555939e30e57]
		??
		??:0
	#13 | /usr/lib/libc.so.6(+0x23850) [0x7fb38894c850]
		??
		??:0
	#14 | /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7fb38894c90a]
		??
		??:0
	#15 | Hyprland(_start+0x25) [0x555939e3e615]
		??
		??:0


Log tail:
[LOG] Callback 55593c5a0c28 -> 55593c5a0c20, layerSurface removed.
[LOG] Callback 55593c5a0cf8 -> 55593c5a0cf0, layerSurface removed.
[LOG] LayerSurface 55593c5785f0 arranged: x: 3072 y: 1704 w: 1920 h: 24 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 55593c4d6220 arranged: x: 3072 y: 528 w: 1920 h: 1200 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 0.000000 0.000000 24.000000
[LOG] Cleanup: destroyed a layersurface
[LOG] Searching for matching rules for wmlauncher_dropdown (title: hyprctl dispatch mov ~)
[LOG] Window rule float -> ^wmlauncher matched 55593c51c170 [hyprctl dispatch mov ~]
[LOG] Window rule size 800 500 -> ^wmlauncher matched 55593c51c170 [hyprctl dispatch mov ~]
[LOG] Window rule center -> ^wmlauncher matched 55593c51c170 [hyprctl dispatch mov ~]
[LOG] Window rule pin -> ^wmlauncher matched 55593c51c170 [hyprctl dispatch mov ~]
[LOG] Window 55593c51c170 set title to hyprctl dispatch mov ~
[LOG]  [hookSystem] New hook event registered: moveWindow
[LOG] Hyprctl: dispatcher movetoworkspacesilent : special
[LOG] Searching for matching rules for wmlauncher_dropdown (title: ~)
[LOG] Window rule float -> ^wmlauncher matched 55593c51c170 [~]
[LOG] Window rule size 800 500 -> ^wmlauncher matched 55593c51c170 [~]
[LOG] Window rule center -> ^wmlauncher matched 55593c51c170 [~]
[LOG] Window rule pin -> ^wmlauncher matched 55593c51c170 [~]
[LOG] Window 55593c51c170 set title to ~
[LOG] Window 55593c51c170 unmapped (class wmlauncher_dropdown)
[LOG]  [hookSystem] New hook event registered: closeWindow
[LOG] Unregistered late callbacks XDG
[LOG] Callback 55593c51c198 -> 55593c51c190, XDG Window Late removed.
[LOG] Callback 55593c51c338 -> 55593c51c330, XDG Window Late removed.
[LOG] Callback 55593c51c470 -> 55593c51c468, XDG Window Late removed.
[LOG] Callback 55593c51c5a8 -> 55593c51c5a0, XDG Window Late removed.
[LOG] Callback 55593c51c540 -> 55593c51c538, XDG Window Late removed.
[LOG] Callback 55593c51c4d8 -> 55593c51c4d0, XDG Window Late removed.
[LOG] Callback 55593c51c610 -> 55593c51c608, XDG Window Late removed.
[LOG] Callback 55593c51c408 -> 55593c51c400, XDG Window Late removed.
[LOG] Framebuffer created, status 36053
[LOG] On closed window, new focused candidate is 55593c5761a0
[LOG] Set keyboard focus to surface 55593c4db8b0, with window name: READACTED READACTED
[LOG] Destroying the SubSurface tree of unmapped window 55593c51c170
[LOG] Callback 55593c4ca028 -> 55593c4ca020, SurfaceTreeNode removed.
[LOG] Callback 55593c4ca090 -> 55593c4ca088, SurfaceTreeNode removed.
[LOG] Callback 55593c4c9fc0 -> 55593c4c9fb8, SurfaceTreeNode removed.
[LOG] SurfaceTree Node removed
[LOG] Callback 55593c51c7b0 -> 55593c51c7a8, Toplevel removed.
[LOG] Callback 55593c51c748 -> 55593c51c740, Toplevel removed.
[LOG] Callback 55593c51c818 -> 55593c51c810, Toplevel removed.
[LOG] Callback 55593c51c8f0 -> 55593c51c8e8, CWLSurface removed.
[LOG] CWLSurface 55593c51c8c0 called destroy()
[LOG] Callback 55593c51c268 -> 55593c51c260, CWindow removed.
[LOG] Window 55593c51c170 destroyed, queueing. (class )
[LOG] Callback 55593c51c200 -> 55593c51c1f8, XDG Window removed.
[LOG] Callback 55593c51c2d0 -> 55593c51c2c8, XDG Window removed.
[LOG] Destroying workspace ID -99
[LOG]  [hookSystem] New hook event registered: destroyWorkspace
```

**Comment by vaxerski** [comment]
fixed in 5484411232a0add23ae0acf9225571fc51e73310

**Comment by foonathan** [comment]
Wow, that was quick.
